tempesta_fw/t/functional/tests/test_cache.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+__author__ = 'Tempesta Technologies'
+__copyright__ = 'Copyright (C) 2016 Tempesta Technologies. (info@natsys-lab.com).'
+__license__ = 'GPL2'
+import sys
+import time
+import types
+
+import conf
+import tfw
+from socket import *
+
+class Test:
+	def __init__(self):
+		self.res = False
+		self.vs_get = b"GET http://localhost:80/ HTTP/1.1\r\n\
+Host: localhost\r\nConnection: Keep-alive\r\nSet-Cookie: session=42\r\n\r\n"
+# to reproduce the bug, a backend (apache) has to return in response double
+# headers.
+		self.apache_cfg = conf.Config('/etc/apache2/apache2.conf',
+					      new=False)
+		self.apache_cfg.add_string('Header add Set-Cookie: \"s=42\"')
+		self.apache_cfg.add_string('Header add Set-Cookie: \"s=42\"')
+		self.cfg = conf.Config("etc/tempesta_fw.conf")
```

Comment:
Ok, I see why do you pass configuration path as an argument to `conf.Config()`. However, there is only one Tempesta config as only one Apache config, it's better to define two appropriate subclasses inherited from `conf` which implement Tempesta and Apached config logic. The code with the sublcasses wild be more readable.